docs/structure-of-a-contract.rst

```diff
@@ -54,6 +54,66 @@ EVM Version
 
 The EVM version can be set with the ``evm-version`` pragma, which is documented in :ref:`evm-version`.
 
+Imports
+=======
+
+Import statements allow you to import :ref:`modules` or :ref:`interfaces` with the ``import`` or ``from ... import`` syntax.
+
+Imports via ``import``
+----------------------
+
+You may import modules (defined in ``.vy`` files) and interfaces (defined in ``.vyi`` or ``.json`` files) via ``import`` statements. You may use plain or ``as`` variants.
+
+.. code-block:: vyper
+
+    # without an alias
+    import my_package.foo
+
+    # with an alias
+    import my_package.foo as bar
+
+Imports via ``from ... import``
+-------------------------------
+
+Using ``from`` you can perform both absolute and relative imports. You may optionally include an alias - if you do not, the name of the interface will be the same as the file.
+
+.. code-block:: vyper
+
+    # without an alias
+    from my_package import foo
+
+    # with an alias
+    from my_package import foo as bar
+
+Relative imports are possible by prepending dots to the contract name. A single leading dot indicates a relative import starting with the current package. Two leading dots indicate a relative import from the parent of the current package:
+
+.. code-block:: vyper
+
+    from . import foo
+    from ..interfaces import baz
+
+.. _searching_for_imports:
+
+Searching For Imports
+-----------------------------
+
+When looking for a file to import, Vyper will first search relative to the same folder as the contract being compiled. It then checks for the file in the provided search paths, in the precedence provided. Vyper checks for the file name with a ``.vy`` suffix first, then ``.vyi``, then ``.json``.
+
+When using the :ref:`vyper CLI <vyper-cli-command>`, the search path defaults to the current working directory, plus the python `syspath <https://docs.python.org/3.11/library/sys.html#sys.path>`_. You can append to the search path with the ``-p`` flag:
+
+::
+
+    $ vyper my_project/contracts/my_contract.vy -p my_project
+
+In the above example, the ``my_project`` folder is set as the root path.
+
+You can additionally disable the behavior of adding the syspath to the search path with the CLI flag ``--disable-sys-path``:
```

Comment:
what's the use case of disabling it?

Reply:
you are a framework and want to control what actually is available in the search path (as opposed to leaving it up to the virtual environment)

Reply:
would it be useful to enable other system paths as well?

Reply:
out of scope of this PR but, no, let's keep it simple and consistent with python workflow. if somebody wants to add node_modules for some reason, let them add it explicitly with `-p`.